homeassistant/components/media_player/squeezebox.py

```diff
@@ -201,11 +201,18 @@ def media_duration(self):
     def media_image_url(self):
         """ Image url of current playing media. """
         if 'artwork_url' in self._status:
-            return self._status['artwork_url']
-        return ('http://{server}:{port}/music/current/cover.jpg?'
-                'player={player}').format(server=self._lms.host,
-                                          port=self._lms.http_port,
-                                          player=self._id)
+            media_url = self._status['artwork_url']
+        else:
+            media_url = ('/music/current/cover.jpg?'
+                         'player={player}&'
+                         'nocache={nocache}').format(
+                             player=self._id,
+                             nocache=hash(self.media_title))
```

Comment:
Thanks! Turns out the cover art can be retrieved both by fetching http://<server>:<port>/music/current/cover.jpg?player=<playerid> and http://<server>:<port>/music/<track_id>/cover.jpg 
I'm switching to the second way, it seems to work fine. Tested both with local content, and streaming content (spotify, radio, podcasts).